tool/tsh/tsh.go

```diff
@@ -72,6 +75,15 @@ var log = logrus.WithFields(logrus.Fields{
 	trace.Component: teleport.ComponentTSH,
 })
 
+type TshConfig struct {
+	ExtraHeaders []ExtraProxyHeaders `yaml:"add_headers"`
+}
+
+type ExtraProxyHeaders struct {
+	Proxy   string            `yaml:"proxy"`
+	Headers map[string]string `yaml:"headers,omitempty"`
+}
```

Comment:
I would isolate structs and methods related to the new config into its own .go file. Looks like `config.go` is already taken, how about `tshconfig.go`?